mindsdb/integrations/handlers/llama_index_handler/llama_index_handler.py

```diff
@@ -0,0 +1,171 @@
+from llama_index.readers.schema.base import Document
+import pandas as pd
+
+from mindsdb.utilities.config import Config
+from mindsdb.integrations.libs.base import BaseMLEngine
+
+from mindsdb.utilities.log import get_log
+
+
+logger = get_log("integrations.llama_index_handler")
+
+
+
+class LlamaIndexHandler(BaseMLEngine):
+    """
+    Integration with the LlamaIndex Python Library
+    """
+    name = 'llama_index'
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+        self.default_index_class  = 'GPTVectorStoreIndex'
+        self.supported_index_class = ['GPTVectorStoreIndex']
+        self.default_reader = 'DFReader'
+        self.supported_reader = ['DFReader','SimpleWebPageReader']
+  
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        """
+        Raises
+        ------
+        ValueError
+            If the query contains an unsupported condition
+        """
+        
+        if 'using' not in args:
+            raise Exception("LlamaIndex engine requires a USING clause! Refer to its documentation for more details.")
+
+
+        if 'openai_api_key' not in args['using']:
+            raise Exception("LlamaIndex engine requires a openai_api_key parameter.Refer to its documentation for more details.")
+
+        if 'index_class' not in args['using']:
+            args['using']['index_class'] = self.default_index_class
+        elif args['using']['index_class'] not in self.supported_index_class:
+            raise Exception(f"Invalid index class argument. Please use one of {self.supported_index_class}")
+
+        if 'reader' not in args['using']:
+            args['using']['reader'] = self.default_reader
+        elif args['using']['reader'] not in self.supported_reader:
+            raise Exception(f"Invalid operation mode. Please use one of {self.supported_reader}")
+
+       
+        documents_df_reader = []
+        documents_url_reader = []
+        pred = None
+        if args['using']['reader'] == 'DFReader':
+            for row in df.itertuples():
+                doc_str = ", ".join([str(entry) for entry in row])
+                documents_df_reader.append(Document(doc_str))  
+            pred = documents_df_reader   
+      
+        elif args['using']['reader'] == 'SimpleWebPageReader':
+
+            if 'source_url_link' not in args['using']:
+                raise Exception("LlamaIndex engine requires a source_url_link parameter.Refer to its documentation for more details.")
+
+            SimpleWebPageReader = download_loader("SimpleWebPageReader")
+            documents_url_reader = SimpleWebPageReader(html_to_text=True).load_data([args['using']['source_url_link']])
+            pred = documents_url_reader 
+
+        else:
+            raise Exception(f"Invalid operation mode. Please use one of {self.supported_reader}.")  
+        
+        self.model_storage.file_set('pred', dill.dumps(pred))
```

Comment:
very nit: I found the naming of "pred" unintuitive.

Reply:
AFAIK (@ea-rus can confirm) there are no limitations on naming with `file_set`, I would suggest perhaps going with `'reader'` instead.

Reply:
yes, no limitations